pkg/controllers/provisioning/provisioner.go

```diff
@@ -136,3 +138,17 @@ func (p *Provisioner) Batch(ctx context.Context) (pods []*v1.Pod) {
 		}
 	}
 }
+
+func (p *Provisioner) RemoveScheduled(ctx context.Context, pods []*v1.Pod) []*v1.Pod {
+	unscheduledPods := []*v1.Pod{}
+	for _, pod := range pods {
+		candidatePod, err := p.launcher.CoreV1Client.Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
```

Comment:
optional:
s/unscheduledPods/unscheduled
s/candidatePod/candidate

nonoptional:
This looks like dependency theft to me. Can you inject in a client.Client instead of relying on the launcher's CoreV1Client?

Reply:
Yeah, this was a WIP, forgot to prepend. Definitely going to clean it up there :D